File: swift/swift/Sources/tutorial/Mob.swift
import SwiftGodot

class Mob: RigidBody2D {
    @Node("AnimatedSprite2D") var animatedSprite2D: AnimatedSprite2D
    @Node("VisibleOnScreenNotifier2D") var visibleOnScreenNotifier2D: VisibleOnScreenNotifier2D

    static let MobGroup = "mobs"

    override func _ready() {
        let mobTypes = self.animatedSprite2D.spriteFrames?.getAnimationNames()
        if let mobTypes = mobTypes {
            // TODO
            let y = GD.randi() % mobTypes.size()
            let x = mobTypes[Int(y)]
            animatedSprite2D.animation = StringName(x)
        }
        self.visibleOnScreenNotifier2D.screenExited.connect {
            self.onScreenExited()
        }
        self.addToGroup(StringName(Mob.MobGroup))
    }

    private func onScreenExited() {
        self.queueFree()
    }
}

File: csharp/Game/Mob.cs
using Godot;
using NodeGetterGenerators;

namespace tutorial;

[GenerateNodeGetter(typeof(AnimatedSprite2D), "AnimatedSprite2D")]
[GenerateNodeGetter(typeof(VisibleOnScreenNotifier2D), "VisibleOnScreenNotifier2D")]
[VerifyNodeGetters("Mob")]
public partial class Mob : RigidBody2D
{
    public const string MobGroup = "mobs";

    public override void _Ready()
    {
        base._Ready();
        var mobTypes = GetNodeAnimatedSprite2D().SpriteFrames.GetAnimationNames();
        GetNodeAnimatedSprite2D().Animation = mobTypes[GD.Randi() % mobTypes.Length];
        GetNodeVisibleOnScreenNotifier2D().ScreenExited += OnScreenExited;
        AddToGroup(MobGroup);
    }

    private void OnScreenExited()
    {
        QueueFree();
    }
}
